packages/cli/src/controller/generate-controller.ts

```diff
@@ -0,0 +1,313 @@
+
+  const casedInputAddress = address && address.toLowerCase();
+
+  dataSources.forEach((ds) => {
+    ds.mapping.handlers.forEach((handler) => {
+      const casedDsAddress = ds.options?.address && ds.options.address.toLowerCase();
+      if (casedDsAddress && casedInputAddress !== casedDsAddress) return;
+      if (Object.keys(handler.filter).includes('topics')) {
+        // topic[0] is the method
+        existingEvents.push((handler.filter as EthereumLogFilter).topics[0]);
+      }
+      if (Object.keys(handler.filter).includes('function')) {
+        existingFunctions.push((handler.filter as EthereumTransactionFilter).function);
+      }
+    });
+  });
+
+  return [
+    filterExistingFragments<EventFragment>(eventFragments, existingEvents),
+    filterExistingFragments<FunctionFragment>(functionFragments, existingFunctions),
+  ];
+}
+
+export async function getManifestData(projectPath: string, manifestPath: string): Promise<Document> {
+  const existingManifest = (await fs.promises.readFile(path.join(projectPath, manifestPath), 'utf8')) as string;
+  return parseDocument(existingManifest);
+}
+
+export async function generateManifest(
+  projectPath: string,
+  manifestPath: string,
+  userInput: UserInput,
+  existingManifestData: Document
+): Promise<void> {
+  const clonedExistingManifestData = existingManifestData.clone();
+  const existingDsNode = existingManifestData.get('dataSources') as YAMLSeq;
+  const dsNode = clonedExistingManifestData.get('dataSources') as YAMLSeq;
+
+  // load any comments
+  dsNode.comment ??= existingDsNode.comment;
+  dsNode.commentBefore ??= existingDsNode.commentBefore;
+
+  dsNode.add(constructDatasources(userInput));
+  await fs.promises.writeFile(path.join(projectPath, manifestPath), clonedExistingManifestData.toString(), 'utf8');
+}
+
+export function constructHandlerProps(methods: [SelectedMethod[], SelectedMethod[]], abiName: string): AbiPropType {
+  const handlers: HandlerPropType[] = [];
+  const [events, functions] = methods;
+
+  functions.map((fn) => {
+    const fnProp: HandlerPropType = {
+      name: `handle${upperFirst(fn.name)}`,
+      argName: 'tx',
+      argType: `${upperFirst(fn.name)}Transaction`,
+    };
+    handlers.push(fnProp);
+  });
+
+  events.map((event) => {
+    const fnProp: HandlerPropType = {
+      name: `handle${upperFirst(event.name)}`,
+      argName: 'log',
+      argType: `${upperFirst(event.name)}Log`,
+    };
+    handlers.push(fnProp);
+  });
+
+  return {
+    name: abiName,
+    handlers: handlers,
+  };
+}
+
+export async function generateHandlers(
+  selectedMethods: [SelectedMethod[], SelectedMethod[]],
+  projectPath: string,
+  abiPath: string
+): Promise<void> {
+  const abiName = parseContractPath(abiPath).name;
+  const abiProps = constructHandlerProps(selectedMethods, abiName);
+
+  try {
+    await renderTemplate(
+      SCAFFOLD_HANDLER_TEMPLATE_PATH,
+      path.join(projectPath, ROOT_MAPPING_DIR, `${abiName}Handlers.ts`),
```

Comment:
What happens if this file exists already?

And if it just overwrites it then do we end up having repeated exports in `src/index.ts`?